### PR TITLE
test(upload): entity manager file signing extension

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Adds unit tests for the entity manager file signing extension in the upload plugin

### CODE REVIEW

**Comment by jhoward1994** [comment]
@Marc-Roig wdyt of this approach to unit testing the extension? I'm going to start working on adding API tests now

File: packages/core/upload/server/services/extensions/content-manager/__tests__/entity-manager.test.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+  describe('signEntityMedia', () => {
+    beforeAll(() => {
+      global.strapi = {
+        plugins: {
+          upload: {},
+        },
+        getModel: jest.fn((uid) => models[uid]),
+      };
+    });
```
Comment (Marc-Roig) [inline packages/core/upload/server/services/extensions/content-manager/__tests__/entity-manager.test.js:75]: Maybe we could avoid importing the `getService` method if we do this:

```suggestion
  describe('signEntityMedia', () => {
    let spySignFileUrls;
    beforeEach(() => {
      spySignFileUrls = jest.fn();
      global.strapi = {
        plugins: {
          upload: {
          	service: () => spySignFileUrls
          },
        },
        getModel: jest.fn((uid) => models[uid]),
      };
    });
```


wdyt?

**Review by Marc-Roig — COMMENTED** [review]
that looks perfect to me 🚀 Nice job.
I have added only a minor comment

**Comment by jhoward1994** [comment]
@Marc-Roig let me know what you think of the latest changes? I've made a minor refactor and added a test case for dynamic zones

File: packages/core/upload/server/services/extensions/content-manager/__tests__/entity-manager.test.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        media_repeatable: media,
+      };
+
+      await signEntityMedia(entity, modelUID);
+      expect(getService).toBeCalledWith('file');
+      expect(spySignFileUrls).toBeCalledTimes(2);
+      expect(spySignFileUrls.mock.calls).toEqual([
+        [media[0], 0],
+        [media[1], 1],
+      ]);
```
Comment (Marc-Roig) [inline packages/core/upload/server/services/extensions/content-manager/__tests__/entity-manager.test.js:108]: minor thing, wdyt of using `toBeCalledWith`?

```suggestion
      expect(spySignFileUrls).toBeCalledWith(media[0], 0);
      expect(spySignFileUrls).toBeCalledWith(media[1], 1);
```

Comment (Marc-Roig) [reply packages/core/upload/server/services/extensions/content-manager/__tests__/entity-manager.test.js:108]: And as the index 0, 1 are not relevant, maybe doing:
```suggestion
      expect(spySignFileUrls).toBeCalledWith(media[0], expect.anything());
      expect(spySignFileUrls).toBeCalledWith(media[1], expect.anything());
```

**Review by Marc-Roig — APPROVED** [review]
LGTM 🚀

**Pull request merged**